test/e2e/gateway_api_test.go

```diff
@@ -158,6 +161,60 @@ func testGatewayAPIObjects(t *testing.T) {
 	}
 }
 
+// testGatewayAPIResourcesProtection verifies that the ingress operator's Validating Admission Policy
+// denies admission requests attempting to modify Gateway API CRDs on behalf of a user
+// who is not the ingress operator's service account.
+func testGatewayAPIResourcesProtection(t *testing.T) {
+	t.Helper()
+
+	var testCRDs []*apiextensionsv1.CustomResourceDefinition
+	for _, name := range crdNames {
```

Comment:
Need to update `crdNames` with GRPCRoute.  It would be beneficial to put the `crdNames` declaration in a shared location. cc @Thealisyed @grzpiotrowski

Reply:
`GRPCRoute` will be added as part of https://github.com/openshift/cluster-ingress-operator/pull/1152. It's not yet mentioned anywhere. However, I can do a helper function in `pkg/manifests/assets/gatewayapi/` package which would return all "desired" CRDs. This way we can use a single function in all controllers and tests.

Reply:
> However, I can do a helper function in pkg/manifests/assets/gatewayapi/ package which would return all "desired" CRDs. This way we can use a single function in all controllers and tests.

I gave this thing another thought. I think that it's better to abstain from using the actual "running" code in the tests. We should explicitly set the list of CRDs we are testing. Otherwise we risk to duplicate (and hide) errors we have in the code.

Reply:
I'm going to hold the rest of my reviews until after #1152 is merged.  Based on the number of changes dependent on it, I feel like I don't want to have to review this again.